### ISSUE TITLE
Bug: [no-loop-func] `using` / `await using` declarations are falsely flagged inside loops

### ISSUE THREAD

**Issue by zaewc** [typescript-eslint/typescript-eslint#12496]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.3&fileType=.tsx&code=GYewTgBAFANgpgFwgSwgXggBgNwogHggGZdkBqMgSggG8AoCCAVwGdkA7AcwjDhZCZgAxnHQROiAEp8BwuFErYGEISHYskwdmIXoAfDxmCRuAPSnm7OAA8ADnCEI4AEwhwwYcBAAW7uHQBfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1vg4%2BNoBmsJmXRRE0aB2iRwYAL4h5QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
for (let i = 0; i < 3; i++) {
  using resource = getResource();
  const fn = () => resource; // unexpected error here
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-loop-func": "error",
  },
};
```

### tsconfig

```jsonc
n/a
```

### Expected Result

no error

`resource` is a `using` declaration, so each loop iteration gets a fresh binding, and the closure can safely capture it

### Actual Result

Function declared in a loop contains unsafe references to variable(s) 'resource'.

**Comment by zaewc** [comment]
I think the cause is that the extension currently only casing `const`

https://github.com/typescript-eslint/typescript-eslint/blob/0cb380fe0d8dd0405f8d25074cf6b283d34d6507/packages/eslint-plugin/src/rules/no-loop-func.ts#L144-L148

**Comment by kirkwaiblinger** [comment]
I can reproduce this, and, yes, without having thought too hard, it seems that the `using` declarations _probably_ should be treated the same as the `const` declarations. But what does the base rule do here? The extension rule should probably just follow the same logic.

**Comment by zaewc** [comment]
The base rule already handles this

https://github.com/eslint/eslint/blob/04174cbfaac1f9555400b5332d69056e6aab9cc6/lib/rules/no-loop-func.js#L18

**Comment by bradzacher** [comment]
Can we just delete the extension rule slash mark it as deprecated? Aren't all of the changes in the extension rule also on the base rule now?

**Comment by zaewc** [comment]
Yeah, looks that way. I ran our full test suite against the base rule and everything passes. The only extra thing we had was the `isTypeReference` check, but the base rule handles those cases fine now without it.

**Comment by zaewc** [comment]
Sorry, I accidentally closed this issue and I don’t have permission to reopen it 😰

**Comment by kirkwaiblinger** [comment]
Deprecate, yes, but can't remove it until the minimum supported ESLint version is raised
